lib/DefinePlugin.js

```diff
@@ -52,15 +64,24 @@ class DefinePlugin {
 					const walkDefinitions = (definitions, prefix) => {
 						Object.keys(definitions).forEach(key => {
 							const code = definitions[key];
+
 							if (
 								code &&
 								typeof code === "object" &&
+								!(code instanceof RuntimeValue) &&
 								!(code instanceof RegExp)
 							) {
 								walkDefinitions(code, prefix + key + ".");
 								applyObjectDefine(prefix + key, code);
 								return;
 							}
+
+							if (code instanceof RuntimeValue && parser.state) {
```

Comment:
@sokra pointed me to use `parser.state.module` to inject file dependencies, but during the unit test I wrote `parser.state` is `undefined`. Feedback is very welcome on how I should change this code or create the right conditions to unit test this.